src/selection-recommendation/recommend.py

```diff
@@ -0,0 +1,130 @@
+import sys
+sys.path.append("../spider/util")
+import itertools
+import pymysql
+import Database
+import time_conflicts_check
+
+DB_NAME = "uoftcourses"
+DB_PATH = "../../database.info"
+
+
+def get_course_data(course_code, campus):
+    """
+    course_code: a string e.g. "CSC148"
+    campus: a string - either "St. George", "Scarborough", or "Mississauga"
+    Returns a list of tuples, with each tuple containing the data of a single section of the specified course
+    """
+    connection = Database.get_connection(DB_PATH, DB_NAME)
+    cursor = connection.cursor()
+    sql = "SELECT * FROM Course WHERE cID like %s AND campus=%s"
+    cursor.execute(sql, ("%" + course_code + "%", campus))
+
+    return(list(cursor.fetchall()))
+
+
+def get_all_possible_course_times(course_code, campus):
```

Comment:
So this first guarantees that the combinations of one course time won't conflict right?

Reply:
Yes

Reply:
After reading your comments I think the 2D array idea is way better.